tests/PHPStan/Analyser/nsrt/bug-12312.php

```diff
@@ -0,0 +1,41 @@
+<?php declare(strict_types = 1);
+
+namespace Bug12312;
+
+use function PHPStan\Testing\assertType;
+
+class HelloWorld
+{
+	/**
+	 * @param lowercase-string $s
+	 */
+	public function sayLowercase(string $s): void
+	{
+		if ($s != '') {
```

Comment:
I think we need another test test for the condition with operands in reversed order